tests/integration/externalapi/test_venafiapi.py

```diff
@@ -25,70 +34,123 @@ def with_random_name(func):
-                                dns_name='{0}.test.saltstack.com'.format(name),
-                                zone='Internet',
-                                password='SecretSauce')
-        self.assertEqual(ret['out'][0], '-----BEGIN RSA PRIVATE KEY-----')
-        self.assertEqual(ret['out'][1], 'Proc-Type: 4,ENCRYPTED')
-        self.assertEqual(ret['out'][-1], '-----END RSA PRIVATE KEY-----')
+    def test_request(self, name):
+        print("Testing Venafi request cert")
+        print("Using venafi config:", self.master_opts['venafi'])
+        cn = '{0}.example.com'.format(name)
+        ret = self.run_run_plus(fun='venafi.request',
+                                minion_id=cn,
+                                dns_name=cn,
+                                key_password='secretPassword',
+                                zone=environ.get('CLOUDZONE'))
+        print("Ret is:\n", ret)
+        cert_output = ret['return'][0]
+        if not cert_output:
+            pytest.fail('venafi_certificate not found in output_value')
 
-    @with_random_name
-    def test_gen_key_without_password(self, name):
-        '''
-        venafi.gen_key
-        '''
-        ret = self.run_run_plus(fun='venafi.gen_key',
-                                minion_id='{0}.test.saltstack.com'.format(name),
-                                dns_name='{0}.test.saltstack.com'.format(name),
-                                zone='Internet')
-        self.assertEqual(ret['out'][0], '-----BEGIN RSA PRIVATE KEY-----')
-        self.assertNotEqual(ret['out'][1], 'Proc-Type: 4,ENCRYPTED')
-        self.assertEqual(ret['out'][-1], '-----END RSA PRIVATE KEY-----')
+        print("Testing certificate:\n", cert_output)
+        cert = x509.load_pem_x509_certificate(cert_output.encode(), default_backend())
+        assert isinstance(cert, x509.Certificate)
+        assert cert.subject.get_attributes_for_oid(NameOID.COMMON_NAME) == [
+            x509.NameAttribute(
+                NameOID.COMMON_NAME, cn
+            )
+        ]
 
-    @with_random_name
-    def test_gen_csr(self, name):
-        '''
-        venafi.gen_csr
-        '''
-        ret = self.run_run_plus(fun='venafi.gen_csr',
-                                minion_id='{0}.test.saltstack.com'.format(name),
-                                dns_name='{0}.test.saltstack.com'.format(name),
-                                country='US', state='Utah', loc='Salt Lake City',
-                                org='Salt Stack Inc.', org_unit='Testing',
-                                zone='Internet', password='SecretSauce')
-        self.assertEqual(ret['out'][0], '-----BEGIN CERTIFICATE REQUEST-----')
-        self.assertEqual(ret['out'][-1], '-----END CERTIFICATE REQUEST-----')
+        pkey_output = ret['return'][1]
+        print("Testing pkey:\n", pkey_output)
+        if not pkey_output:
+            pytest.fail('venafi_private key not found in output_value')
```

Comment:
Same thing here - `load_pem_private_key` will fail if there's a problem with `pkey_output`

Reply:
Same as before, I'd prefer to keep it for easier understanding

Reply:
And same solution as before 😄 - it would be fine to change it to `assert pkey_output is not None, 'venafi_private...'`